src/rdkafka_assignor.c

```diff
@@ -558,3 +559,403 @@ int rd_kafka_assignors_init (rd_kafka_t *rk, char *errstr, size_t errstr_size) {
+                                             metadata.topic_cnt);
+                memset(metadata.topics, 0,
+                       sizeof(*metadata.topics) * metadata.topic_cnt);
+                for (it = 0 ; it < metadata.topic_cnt ; it++) {
+                        metadata.topics[it].topic =
+                                (char *)tests[i].topics[it].name;
+                        metadata.topics[it].partition_cnt =
+                                tests[i].topics[it].partition_cnt;
+                        metadata.topics[it].partitions = NULL; /* Not used */
+                }
+
+                /* Create members */
+                members = rd_alloca(sizeof(*members) * tests[i].member_cnt);
+                memset(members, 0, sizeof(*members) * tests[i].member_cnt);
+
+                for (im = 0 ; im < tests[i].member_cnt ; im++) {
+                        rd_kafka_group_member_t *rkgm = &members[im];
+                        rkgm->rkgm_member_id =
+                                rd_kafkap_str_new(tests[i].members[im].name,
+                                                  -1);
+                        rkgm->rkgm_group_instance_id =
+                                rd_kafkap_str_new(tests[i].members[im].name,
+                                                  -1);
+                        rd_list_init(&rkgm->rkgm_eligible,
+                                     tests[i].members[im].topic_cnt, NULL);
+
+                        rkgm->rkgm_subscription =
+                                rd_kafka_topic_partition_list_new(
+                                        tests[i].members[im].topic_cnt);
+                        for (it = 0; it < tests[i].members[im].topic_cnt; it++)
+                                rd_kafka_topic_partition_list_add(
+                                        rkgm->rkgm_subscription,
+                                        tests[i].members[im].topics[it],
+                                        RD_KAFKA_PARTITION_UA);
+
+                        rkgm->rkgm_userdata = NULL;
+
+                        rkgm->rkgm_assignment =
+                                rd_kafka_topic_partition_list_new(
+                                        rkgm->rkgm_subscription->size);
+                }
+
+                /* For each assignor verify that the assignment
+                 * matches the expection set out in the test case. */
+                for (ie = 0 ; ie < tests[i].expect_cnt ; ie++) {
+                        rd_kafka_resp_err_t err;
+                        char errstr[256];
+
+                        RD_UT_SAY("Test case %s: %s assignor",
+                                  tests[i].name,
+                                  tests[i].expect[ie].protocol_name);
+
+                        /* Run assignor */
+                        err = rd_kafka_assignor_run(
+                                rk->rk_cgrp,
+                                tests[i].expect[ie].protocol_name,
```

Comment:
heads up that i've changed this to be a pointer to the assignor in the change for stateful assignors. also, `rk->rk_cgrp` maintains this state so it's kind of superfluous to specify it separately. anyway fothcoming change.